### PR TITLE
Optimize Battery Icon Calculation and State Handling

### LINKED ISSUE #270: Battery status goes to zero when plugging/unplugging cable
### What Happened?

While plugging in or unplugging the power cable, the battery status temporarily goes to 0/critical for a few seconds:

image

**UPDATE**:
The same thing happens every time a Bluetooth device is connected.

### Steps to Reproduce

1. With battery 100% charged, unplug the cable
2. Icon goes to 0/critical while percentage stays at 100%
3. After a few seconds, the correct icon appears again
4. Plug in the cable, icon goes to 0/critical again

### Expected Behavior

The icon should show the correct percentage

### OS Version

8.x (Early Access)

### Session Type

Classic Session (X11, This is the default)

### Software Version

Latest release (I have run all updates)

### Log Output

_No response_

### Hardware Info

```
         eeeeeeeeeeeeeeeee             wout@Framey
      eeeeeeeeeeeeeeeeeeeeeee          -----------
    eeeee  eeeeeeeeeeee   eeeee        OS: elementary OS circe 8 x86_64
  eeee   eeeee       eee     eeee      Host: Laptop (12th Gen Intel Core) (A8)
 eeee   eeee          eee     eeee     Kernel: Linux 6.8.0-48-generic
eee    eee            eee       eee    Uptime: 12 hours, 7 mins
eee   eee            eee        eee    Packages: 1811 (dpkg), 18 (flatpak-system), 18 (flatpak-user), 7 (sn)
ee    eee           eeee       eeee    Shell: bash 5.2.21
ee    eee         eeeee      eeeeee    Display (BOE095F): 3000x2000 @ 60 Hz in 13" [Built-in]
ee    eee       eeeee      eeeee ee    DE: Pantheon
eee   eeee   eeeeee      eeeee  eee    WM: Mutter(Gala) (X11)
eee    eeeeeeeeee     eeeeee    eee    Theme: Yaru [GTK3], io.elementary.stylesheet.blueberry [GTK4]
 eeeeeeeeeeeeeeeeeeeeeeee    eeeee     Icons: Yaru [GTK3], elementary [GTK4]
  eeeeeeee eeeeeeeeeeee      eeee      Font: Inter (9pt) [GTK4]
    eeeee                 eeeee        Cursor: elementary
      eeeeeee         eeeeeee          Terminal: x-terminal-emulator
         eeeeeeeeeeeeeeeee             CPU: 12th Gen Intel(R) Core(TM) i7-1280P (20) @ 4.80 GHz
                                       GPU: Intel Iris Xe Graphics @ 1.45 GHz [Integrated]
                                       Memory: 7.24 GiB / 62.51 GiB (12%)
                                       Swap: 0 B / 3.80 GiB (0%)
                                       Disk (/): 70.54 GiB / 910.86 GiB (8%) - ext4
                                       Disk (/media/wout/1TB Framework): 168.41 GiB / 913.59 GiB (18%) - ex4
                                       Local IP (wlp166s0): 192.168.1.253/24
                                       Battery (Framewo): 93% (2 hours, 56 mins remaining) [Discharging]
                                       Locale: en_US.UTF-8
```

### PR DESCRIPTION
**Pull request by alainm23**
Fixed #270

### CODE REVIEW

File: src/Services/Device.vala [hunk 1]
```diff
@@ -231,29 +231,35 @@ public class Power.Services.Device : Object {
...
+        // Add charging state if applicable
         if (is_charging) {
             icon_name += "-charging";
-        } else if (time_to_empty >= 0 && time_to_empty < 15 * 60) {
+        } else if (percentage == 0) {
+            // Only set battery-0 when the battery is empty (0%)
             icon_name = "battery-0";
+        } else if (time_to_empty >= 0 && time_to_empty < 15 * 60 && percentage > 0) {
+            // Prevent showing battery-0-symbolic if there's still battery left
+            icon_name = "battery-%i".printf (rounded_percentage);
```
Comment (danirabbit) [inline src/Services/Device.vala:262]: This will result in missing icon for those ranges below, I think it's 20

Comment (alainm23) [reply src/Services/Device.vala:262]: What would be the case where it could fail? In a battery range between 0 and 20?

Comment (danirabbit) [reply src/Services/Device.vala:262]: Yeah IIRC the icons go from 0 to 20 and then every 5 or 10 after that. I can't remember and I'm AFK. But make sure to double check that :)

Comment (alainm23) [reply src/Services/Device.vala:260]: I did the test with percentages less than 20% and they all work correctly.

image

**Review by leonardo-lemos — APPROVED** [review]
LGTM

**Pull request merged**